builder/dockerfile/builder.go

```diff
@@ -77,6 +103,36 @@ func (bm *BuildManager) Build(ctx context.Context, config backend.BuildConfig) (
 	return newBuilder(ctx, builderOptions).build(source, dockerfile)
 }
 
+func (bm *BuildManager) initializeClientSession(ctx context.Context, cancel func(), options *types.ImageBuildOptions) (builder.Source, error) {
```

Comment:
When other features will be based on client session (such as registry authent, secrets,...) it would be good to split this one between `initializeClientSession` (returning a `session.Caller`) and `initializeRemoteContext` for initializing the `builder.Source`.